site/docs/v1/tech/migration-guide/cognito.adoc

```diff
@@ -0,0 +1,708 @@
+
+Alternatively, you can do a bulk migration and force everyone to reset their passwords. This option is discussed below, but the primary focus of this guide is enabling you to migrate your users from Cognito without requiring any password resets.
+
+== Planning Considerations
+
+=== Slow Migration or Bulk Migration
+
+To preserve your user's passwords, you need to perform a slow migration. Users log in to FusionAuth with their Cognito credentials, and FusionAuth transparently migrates their data. Slow migrations in FusionAuth use Connectors, a paid feature.
+
+If, on the other hand, resetting user passwords is acceptable, a <<Bulk Migration>> can work for you. Review that section for more details on the required steps. You may also perform a bulk migration after a slow migration has run for a while. Active users can be transparently migrated and infrequent users may not mind resetting their password.
+
+You can learn more about the link:/docs/v1/tech/migration-guide/general/#types-of-migrations[types of migrations that FusionAuth supports here].
+
+=== Mapping User Attributes
+
+include::docs/v1/tech/migration-guide/_mapping-user-attributes.adoc[]
+
+=== One Tenant or Many
+
+In Cognito, users are placed in https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools.html[User Pools]. FusionAuth has a similar concept called link:/docs/v1/tech/core-concepts/tenants/[Tenants]. Both of these include data about users, applications and other configuration.
+
+Each tenant in FusionAuth is a distinct user space. You may choose to merge multiple Cognito User Pools into one FusionAuth tenant or keep them separate.
+
+Learn more about link:/docs/v1/tech/guides/multi-tenant/[FusionAuth tenants].
+
+=== Identity Pools
+
+In Cognito, external identity providers are managed using https://docs.aws.amazon.com/cognito/latest/developerguide/external-identity-providers.html[Identity Pools]. With FusionAuth, these are called link:/docs/v1/tech/identity-providers/[Identity Providers]. 
+
+Review the supported FusionAuth link:/docs/v1/tech/identity-providers/[Identity Providers] to ensure any you need are supported. At this time, all Cognito external identity providers are supported by FusionAuth, except `Login with Amazon`.
+
+If not supported explicitly, a provider may work with an OIDC or SAML connection. Otherwise, please open a https://github.com/fusionauth/fusionauth-issues/[feature request].
+
+[NOTE.note]
+====
+Importing Cognito users stored in an external identity provider is more straightforward because you can use the Import API; you don't have to perform a slow migration. Many of the steps in <<Bulk Migration>> will apply.
+====
+
+To retrieve the user information, use the approach documented in the <<Extract All Users>> section.
+
+include::docs/v1/tech/migration-guide/_social-login-migration.adoc[]
```

Comment:
the langauge in this doc has some grammatical clean up needed:

"However, if the user has a social provider email address (richard@gmail.com) which differs from the email address that was used to sign up for your application and which you imported to FusionAuth (richard@fusionauth.io). (The same holds true for usernames.) This will cause two accounts to be created."

Reply:
Thanks, good catch